src/crimson/os/seastore/cache.cc

```diff
@@ -1289,21 +1340,27 @@ Cache::get_next_dirty_extents_ret Cache::get_next_dirty_extents(
   size_t max_bytes)
 {
   LOG_PREFIX(Cache::get_next_dirty_extents);
+  if (dirty.empty()) {
+    DEBUGT("max_bytes={}B, seq={}, dirty is empty",
+           t, max_bytes, seq);
+  } else {
+    DEBUGT("max_bytes={}B, seq={}, dirty_from={}",
+           t, max_bytes, seq, dirty.begin()->get_dirty_from());
+  }
   std::vector<CachedExtentRef> cand;
   size_t bytes_so_far = 0;
   for (auto i = dirty.begin();
        i != dirty.end() && bytes_so_far < max_bytes;
        ++i) {
     if (i->get_dirty_from() != journal_seq_t() && i->get_dirty_from() < seq) {
```

Comment:
Seems to me dirty extents always need a valid dirty_from. If it is correct, should we instead assert that `i->get_dirty_from() != journal_seq_t()` is always true?

Reply:
Yeah, I think that's right.